core/opendaq/component/include/opendaq/status_event_args.h

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright 2022-2023 Blueberry d.o.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <coretypes/coretypes.h>
+#include <coretypes/event_args.h>
+
+BEGIN_NAMESPACE_OPENDAQ
+
+/*#
+ * [interfaceSmartPtr(IEventArgs, EventArgsPtr, "<coretypes/event_args_ptr.h>")]
+ * [interfaceSmartPtr(IEnumeration, EnumerationPtr, "<coretypes/enumeration_ptr.h>")]
+ */
+
+/*!
+ * @ingroup opendaq_component_status_container
+ * @addtogroup opendaq_status_events Component Status Event Args
+ * @{
+ */
+
+/*!
+ * @brief Arguments object that provides a name and a value of the changed component status.
+ *
+ * Status events are triggered whenever there is a change in the status of the openDAQ Component.
+ * Each status has a unique name and represents an actual status related to the Component,
+ * such as connection status, synchronization status, etc. The Status value is determined by the
+ * Enumeration object.
+ *
+ * The Status Event object can be obtained from the Component status container object, which is
+ * accessible through the `getStatus()` call for any component within the openDAQ tree structure.
+ */
+DECLARE_OPENDAQ_INTERFACE(IStatusEventArgs, IEventArgs)
+{
+    /*!
+     * @brief Gets the name of changed Component status.
+     * @param[out] name The name of changed Component status.
+     */
+    virtual ErrCode INTERFACE_FUNC getStatusName(IString** name) = 0;
+
+    /*!
+     * @brief Gets the value of changed Component status.
+     * @param[out] value The value of changed Component status.
+     */
+    virtual ErrCode INTERFACE_FUNC getStatusValue(IEnumeration** value) = 0;
+};
```

Comment:
This should use the CoreEventArgs interface and mechanism. The StatusChanged on a component should be a core event.